manage_containers.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python3
+
+"""
+This module provides functionality to manage Docker containers
+that are configured for building kernels with specific versions
+of GCC and Clang compilers.
+"""
+
+import os
+import subprocess
+import sys
+import argparse
+
+supported_compilers = ['gcc-4.9', 'gcc-5', 'gcc-6', 'gcc-7', 'gcc-8', 'gcc-9',
+                       'gcc-10', 'gcc-11', 'gcc-12', 'gcc-13', 'gcc-14',
+                       'clang-5.0', 'clang-6.0', 'clang-7', 'clang-8',
+                       'clang-9', 'clang-10', 'clang-11', 'clang-12',
+                       'clang-13', 'clang-14', 'clang-15', 'clang-16', 'clang-17',
+                       'all']
+
+class Container:
+    """
+    Represents a Docker container configured for kernel building
+
+    Class Attributes:
+        sudo (str): represents sudo command
+        quiet (bool): enable quiet option in Docker
+
+    Container Attributes:
+        gcc (str): Version of GCC compiler
+        clang (str): Version of Clang compiler (optional)
+        ubuntu (str): Version of Ubuntu
+        id (str): Docker image id
+    """
+
+    sudo = ''
+    quiet = False
+
+    def __init__(self, gcc_version, clang_version, ubuntu_version):
+        self.gcc = gcc_version
+        self.clang = clang_version
+        self.ubuntu = ubuntu_version
+        self.id = self.check()
+
+    def add(self):
+        """Builds the Docker container with the specified compilers"""
+        build_args = ['--build-arg', f'GCC_VERSION={self.gcc}',
+                      '--build-arg', f'CLANG_VERSION={self.clang}',
+                      '--build-arg', f'UBUNTU_VERSION={self.ubuntu}',
+                      '--build-arg', f'UNAME={os.getlogin()}',
+                      '--build-arg', f'UID={os.getuid()}',
+                      '--build-arg', f'GID={os.getgid()}',
+                      '-t', f'kernel-build-container:gcc-{self.gcc}',
+                      '-t', f'kernel-build-container:clang-{self.clang}'
+        ]
+
+        if self.quiet:
+            build_args += ['-q']
+        subprocess.run([self.sudo, 'docker', 'build', *build_args, '.'],
+                        text=True, check=True)
+        self.check()
+
+    def rm(self):
+        """Removes the Docker container if it exists and not running"""
+        running = subprocess.run(f"{Container.sudo} docker ps | "
```

Comment:
This is still relevant for other cases.

Reply:
My bad, I missed some. Now, double quotations are only in the comments and one subprocess block, which I think needs to be refactored first 

https://github.com/a13xp0p0v/kernel-build-containers/pull/22/commits/be0388195dc689bfbf4b0c6261c53952a766fcf0.

Reply:
So, because of this refactor:

https://github.com/a13xp0p0v/kernel-build-containers/pull/22/commits/76a512ca88e5b286ac4240aa94c6a0b26cae04aa

there are no more double quotation marks left (except comments).